test/create-user.test.ts

```diff
@@ -0,0 +1,68 @@
+import { expect } from 'chai';
+import { getConnection } from 'typeorm';
+import { UserEntity } from '../src/data/entity/user.entity';
+import { GraphQLServer } from '../src/server';
+import bcrypt from 'bcrypt';
+
+const url = `http://localhost:3000/`;
+const request = require('supertest')(url);
+
+before(async () => {
+  try {
+    const server = new GraphQLServer();
+    await server.startServer();
+  } finally {
+    const connection = getConnection();
+    await connection.synchronize();
+  }
+});
+after(async () => {
+  const connection = getConnection();
+  await connection.dropDatabase();
+  await connection.close();
+});
+
+describe('User endpoint', function () {
+  it('returns the data from the created a user', (done) => {
+    request
+      .post('graphql')
+      .send({
+        query:
+          'mutation { createUser(userInput: { name: "Machado de Assis", email: "machado@assis.com", password: "machado45515" , dateOfBirth: "06-21-1839" }){ id, name, email, dateOfBirth} }',
+      })
+      .end((_err: any, res: { body: { data: any } }) => {
+        expect(res.body.data.createUser).to.include({
+          name: 'Machado de Assis',
+          email: 'machado@assis.com',
+        });
+        expect(res.body.data.createUser.id).to.exist;
+        expect(res.body.data.createUser.dateOfBirth).to.exist;
+
+        done();
+      });
+  }),
+    it('saves the user in the database', async () => {
+      await createUser();
+      const connection = getConnection();
+      const userRepository = connection.getRepository(UserEntity);
+      const user = await userRepository.findOne();
+
+      expect(user?.name).to.be.equal('Machado de Assis');
+    }),
+    it('saves a hashed password', async () => {
+      await createUser();
+      const connection = getConnection();
+      const userRepository = connection.getRepository(UserEntity);
+      const user = await userRepository.findOne();
+
+      expect(bcrypt.compare('machado45515', user?.password));
+    });
+});
+
```

Comment:
Acho que dá pra organizar um pouco melhor o teste. Alguns pontos:

1. Esses 3 casos de teste poderiam ser resumidos em 1 só, porque os checks que vc está fazendo podem ser feitos apenas em um dos testes. Faz sentido?
2. O teste poderia estar um pouco mais completo: você testou a resposta da mutation no primeiro it, mas ainda assim não de forma tão completa (sugiro o to.be.deep.eq, dá uma lida sobre ele). E nos outros 2 testes restou apenas o nome e o password que foram salvos no banco, mas não os outros campos.

Segue [um doc](https://www.notion.so/taqtile/Boas-pr-ticas-para-testes-66f8b1e317be4d88bde55604d45f8fe1) que pode te ajudar.